### PR TITLE
Fix crash in ProjectedVolume component

### LINKED ISSUE #5461: Cannot read properties of null (reading 'map')
**Describe the bug**
If I click on some of my pods the application crashes. This seems to be related to the postgres pods created by crunchy pgo v5 (but will be "something" to do with the pods. 

**To Reproduce**
1. Click on the "wrong pod"
2. Boom - "App crash at /pods"

**Expected behavior**
I see the pod details

**Environment (please complete the following information):**
- Lens Version: 5.5.0
- OS: Windows
- Installation method (e.g. snap or AppImage in Linux): Auto upgrade

**Logs:**
When you run the application executable from command line you will see some logging output. Please paste them here:
```
Component stack:


    at Projected (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:536235:20)
    at VolumeVariant (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:536605:20)
    at div
    at observerComponent (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:266183:16)
    at div
    at PodDetails (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:536682:9)
    at div
    at div
    at div
    at Animate (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:573737:9)
    at NonInjectedDrawer (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:579968:9)
    at KubeObjectDetails (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:584458:9)
    at ErrorBoundary (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:582139:9)
    at NonInjectedClusterFrame (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/chunks/888.js:2924:9)
    at ThemeProvider (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:127160:24)
    at Router (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:454676:30)

Error stack:

TypeError: Cannot read properties of null (reading 'map')
    at Projected (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:536239:81)
    at Ch (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:453189:137)
    at ck (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:453299:460)
    at bk (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:453282:347)
    at ak (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:453282:278)
    at Tj (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:453282:138)
    at Lj (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:453275:163)
    at http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:453155:115
    at exports.unstable_runWithPriority (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:467222:343)
    at gg (http://d53dcd08fb0a052059c55a0603ecf0ba.localhost:51340/build/Lens.js?266aed5a965262b158f7:453154:325)
```
**Kubeconfig:**

```
Kubeconfig was fine 10 minutes ago.
```

**Additional context**
I am using GKE

### PR DESCRIPTION
**Pull request by Nokel81**
- Also update the typings

Signed-off-by: Sebastian Malton <sebastian@malton.name>

fixes https://github.com/lensapp/lens/issues/5461

### CODE REVIEW

File: src/renderer/components/+workloads-pods/details/volumes/variants/__snapshots__/projected.test.tsx.snap [hunk 1]
```diff
@@ -0,0 +1,229 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`<Projected /> renders 1`] = `
```
Comment (Nokel81) [inline src/renderer/components/+workloads-pods/details/volumes/variants/__snapshots__/projected.test.tsx.snap:3]: I know that these are not exhaustive tests. We can add some more in the future I think.

File: src/common/k8s-api/endpoints/pod.api.ts [hunk 2]
```diff
@@ -445,46 +445,62 @@ export interface PortworxVolumeSource {
...
+
+export interface ObjectFieldSelector {
+  fieldPath: string;
+  apiVersion?: string;
+}
+
+export interface ResourceFieldSelector {
+  resource: string;
+  containerName?: string;
+  divisor?: string;
```
Comment (jim-docker) [inline src/common/k8s-api/endpoints/pod.api.ts:468]: added for completeness?

File: src/common/k8s-api/endpoints/pod.api.ts [hunk 3]
```diff
@@ -445,46 +445,62 @@ export interface PortworxVolumeSource {
...
+  items?: DownwardAPIVolumeFile[];
+}
+
+export interface SecretProjection {
+  name: string;
+  items?: KeyToPath[];
+  optional?: boolean;
+}
+
+export interface ServiceAccountTokenProjectio {
```
Comment (jim-docker) [inline src/common/k8s-api/endpoints/pod.api.ts]: nit:
```suggestion
export interface ServiceAccountTokenProjection {
```

File: src/renderer/components/+workloads-pods/details/volumes/variants/projected.tsx [hunk 4]
```diff
@@ -10,12 +10,14 @@ import type { VolumeVariantComponent } from "../variant-helpers";
...
     <>
-      <DrawerItem name="Default Mount Mode">
-        {`0o${defaultMode.toString(8)}`}
-      </DrawerItem>
+      {typeof defaultMode === "number" && (
+        <DrawerItem name="Default Mount Mode">
+          {`0o${defaultMode.toString(8)}`}
+        </DrawerItem>
+      )}
       <DrawerItem name="Sources">
```
Comment (jim-docker) [inline src/renderer/components/+workloads-pods/details/volumes/variants/projected.tsx:19]: Should we show this at all if `sources` is null/undefined? (as long as we're consistent with what is done elsewhere)

**Review by jim-docker — DISMISSED** [review]
More tests are always good but just a reminder that bug fix PRs can be reviewed more quickly if they don't include refactorings and enhancements.

File: src/common/k8s-api/endpoints/pod.api.ts (hunk 2 above)
Comment (Nokel81) [reply src/common/k8s-api/endpoints/pod.api.ts:468]: Yes, my source is `@kubernetes/node`. I checked what their types were and they said that the `sources` field was optional, but to make the types easier to read I split out the field types.

File: src/renderer/components/+workloads-pods/details/volumes/variants/projected.tsx (hunk 4 above)
Comment (Nokel81) [reply src/renderer/components/+workloads-pods/details/volumes/variants/projected.tsx:19]: I would say yes, but we can always change it in the future.

**Comment by Nokel81** [comment]
> More tests are always good but just a reminder that bug fix PRs can be reviewed more quickly if they don't include refactorings and enhancements.

Fair enough, I wanted to show why the fix was applicable.

File: src/renderer/components/+workloads-pods/details/volumes/variants/projected.tsx [hunk 5]
```diff
@@ -25,9 +27,12 @@ export const Projected: VolumeVariantComponent<"projected"> = (
                   </DrawerItem>
                   <DrawerItem name="Items">
                     <ul>
-                      {secret.items?.map(({ key, path }) => (
+                      {secret.items?.map(({ key, path, mode }) => (
                         <li key={key}>
                           {`${key}⇢${path}`}
+                          {typeof mode === "number" && (
+                            ` (0o${mode.toString(8)})`
```
Comment (ixrock) [inline src/renderer/components/+workloads-pods/details/volumes/variants/projected.tsx]: _nit:_ better to move to external method for formatting `mode`  (duplicated 2 times in the file already)

**Review by ixrock — DISMISSED** [review]

File: src/renderer/components/+workloads-pods/details/volumes/variants/projected.tsx (hunk 5 above)
Comment (Nokel81) [reply src/renderer/components/+workloads-pods/details/volumes/variants/projected.tsx]: Good point, will do

File: src/common/k8s-api/endpoints/pod.api.ts (hunk 2 above)
Comment (jim-docker) [reply src/common/k8s-api/endpoints/pod.api.ts:468]: yes, but divisor is new, and not used in this PR

**Review by ixrock — APPROVED** [review]

**Review by aleksfront — APPROVED** [review]
LGTM

**Pull request merged**